backend/routes/webhooks.ts

```
@@ -0,0 +1,122 @@
+                    // For now, we focus on UPDATING existing records based on stripe_subscription_id.
+                    // The initial INSERT with user_id needs to happen elsewhere (e.g., after successful checkout).
+
+                    const upsertData = {
+                        stripe_subscription_id: subscription.id,
+                        stripe_customer_id: subscription.customer as string,
+                        status: subscription.status as SubscriptionStatus, // Cast to our manual type
+                        current_period_end: new Date(subscription.current_period_end * 1000).toISOString(),
+                        updated_at: new Date().toISOString()
+                        // user_id is NOT included here as we can't reliably get it from the event alone
+                        // for *new* subscriptions without prior setup during checkout.
+                    };
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)